### PR TITLE
esys: remove trailing zeros in auth value.

### PR DESCRIPTION
**Pull request by JuergenReppSIT**
When TPM Calulates the HMAC trailing zeros are rmoved. Therfore the
trailing zeros are removed in Esys_TR_SetAuth, Esys_Create, Esys_CreateLoded,
Esys_NV_DefineSpace, and Esys_CreatePrimary.
The auth value is also adapted in the ChangeAuth programs

Addresses  #2664

Signed-off-by: Juergen Repp <juergen_repp@web.de>

### CODE REVIEW

**Comment by williamcroberts** [comment]
@JuergenReppSIT we updated the bug tracker #2660 with complete trigger examples, any way you could respin this to get it to fail? I backed your patch out and ran the test and couldn't get a failure. Perhaps I didn't do something properly locally. Don't we also need to handle the auth values when we create objects (Esys_Create, Esys_CreateLoaded, Esys_CreatePrimary, Esys_NV_DefineSpace) since they get remembered?

**Comment by JuergenReppSIT** [comment]
@williamcroberts sorry I forgot, that Esys_Create does not create an esys object. The object will be created by Esys_Load. So the call of Esys_TR_SetAuth is needed for keys. 
I just added the zero stripping to the function which computes the hash for long password. The integration tests work. 
I will remove the draft flag after the following additional changes:
* The integration test is extended with the example which did produce the failure.
* The functions Esys_ChangeAuth, Esys_NV_ChangeAuth , and Esys_ObjectChangeAuth. do not call the function which adapts long passwords.

File: src/tss2-esys/api/Esys_HierarchyChangeAuth.c [hunk 1]
```diff
@@ -175,7 +175,9 @@ Esys_HierarchyChangeAuth_Async(
     /* Check input parameters */
     r = check_session_feasibility(shandle1, shandle2, shandle3, 1);
     return_state_if_error(r, _ESYS_STATE_INIT, "Check session usage");
+
     store_input_parameters(esysContext, authHandle, newAuth);
+    iesys_strip_trailing_zeros(&esysContext->in.HierarchyChangeAuth.newAuth);
```
Comment (williamcroberts) [inline src/tss2-esys/api/Esys_HierarchyChangeAuth.c:180]: any reason to not do this in store_input_parameters?

File: src/tss2-esys/api/Esys_ObjectChangeAuth.c [hunk 2]
```diff
@@ -163,12 +179,22 @@ Esys_ObjectChangeAuth_Async(
...
 
+    store_input_parameters(esysContext, objectHandle, newAuth);
+
     /* Retrieve the metadata objects for provided handles */
     r = esys_GetResourceObject(esysContext, objectHandle, &objectHandleNode);
     return_state_if_error(r, _ESYS_STATE_INIT, "objectHandle unknown.");
     r = esys_GetResourceObject(esysContext, parentHandle, &parentHandleNode);
     return_state_if_error(r, _ESYS_STATE_INIT, "parentHandle unknown.");
 
+    if (objectHandleNode->rsrc.rsrcType == IESYSC_KEY_RSRC) {
```
Comment (williamcroberts) [inline src/tss2-esys/api/Esys_ObjectChangeAuth.c:190]: what about NV objects?

File: src/tss2-esys/api/Esys_HierarchyChangeAuth.c (hunk 1 above)
Comment (JuergenReppSIT) [reply src/tss2-esys/api/Esys_HierarchyChangeAuth.c:180]: A good hint. The spec states:

This command does not change the authorization of the TPM-resident object on which it operates.
Therefore, the old authValue (of the TPM-resident object) is used when generating the response HMAC
key if required. 

So it makes no sense to store the auth value in the esys object. I will remove this change.

File: src/tss2-esys/api/Esys_ObjectChangeAuth.c (hunk 2 above)
Comment (JuergenReppSIT) [reply src/tss2-esys/api/Esys_ObjectChangeAuth.c:190]: The spec states:
NOTE 3 If an NV Index is to have a new authorization, it is done with TPM2_NV_ChangeAuth().

File: src/tss2-esys/api/Esys_HierarchyChangeAuth.c (hunk 1 above)
Comment (cplappert) [reply src/tss2-esys/api/Esys_HierarchyChangeAuth.c:180]: I think @williamcroberts wants the "iesys_strip_trailing_zeros" command to be implemented in the "store_input_parameters" function. This makes also sense to me. 
@JuergenReppSIT: Could you do this?

**Review by AndreasFuchsTPM — APPROVED** [review]

**Pull request merged**